indigo_app/views/tasks.py

```diff
@@ -287,6 +290,21 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class TaskEditLabelsView(SingleTaskViewBase, FormView, SingleObjectMixin):
```

Comment:
If you make this inherit from `UpdateView`, then it should Just Work because it knows how to apply changes to an existing object. You'd need to make the form class a ModelForm, but it only needs the labels field.

Probably not vastly different to what you have, but it does use django standards a bit more.

Reply:
also double check the right perms are required (change task, and the country perm)

Reply:
thanks! I knew there was something fishy about it :)